go/osagent/osagent.go

```diff
@@ -645,3 +745,609 @@ func AbortSeed(seedId string) error {
+		return "", log.Errore(err)
+	}
+	switch seedMethod {
+	case "xtrabackup":
+		cmd = fmt.Sprintf("xtrabackup --backup --user=%s --password=%s --port=%d --parallel=%d --target-dir=%s --databases='%s'",
+			config.Config.MySQLTopologyUser, config.Config.MySQLTopologyPassword, config.Config.MySQLPort, config.Config.XtrabackupParallelThreads, BackupFolder, strings.Replace(databases, ",", " ", -1))
+	case "mydumper":
+		cmd = fmt.Sprintf("mydumper --user=%s --password=%s --port=%d --threads=%d --outputdir=%s --triggers --events --routines --regex='(%s)'",
+			config.Config.MySQLTopologyUser, config.Config.MySQLTopologyPassword, config.Config.MySQLPort, config.Config.MyDumperParallelThreads, BackupFolder, strings.Replace(databases, ",", "\\.|", -1)+"\\.")
+		if config.Config.CompressLogicalBackup {
+			cmd += fmt.Sprintf(" --compress")
+		}
+		if config.Config.MyDumperRowsChunkSize != 0 {
+			cmd += fmt.Sprintf(" --rows=%d", config.Config.MyDumperRowsChunkSize)
+		}
+	case "mysqldump":
+		if databases == "" {
+			cmd = fmt.Sprintf("mysqldump --user=%s --password=%s --port=%d --single-transaction --master-data=1 --routines --events --triggers --all-databases",
+				config.Config.MySQLTopologyUser, config.Config.MySQLTopologyPassword, config.Config.MySQLPort)
+		} else {
+			cmd = fmt.Sprintf("mysqldump --user=%s --password=%s --port=%d --single-transaction --master-data=1 --routines --events --triggers --databases %s",
+				config.Config.MySQLTopologyUser, config.Config.MySQLTopologyPassword, config.Config.MySQLPort, strings.Replace(databases, ",", " ", -1))
+		}
+		if config.Config.CompressLogicalBackup {
+			cmd += fmt.Sprintf(" | gzip > %s", path.Join(BackupFolder, mysqlbackupCompressedFileName))
+		} else {
+			cmd += fmt.Sprintf(" > %s", path.Join(BackupFolder, mysqlbackupFileName))
+		}
+	}
+	err = commandRun(
+		fmt.Sprintf(cmd),
+		func(cmd *exec.Cmd) {
+			activeCommands[seedId] = cmd
+			log.Debug("StartLocalBackup command completed")
+		})
+	if err != nil {
+		return "", log.Errore(err)
+	}
+	return BackupFolder, err
+}
+
+func backupMySQLUsers(seedId string) (err error) {
+	if len(config.Config.MySQLBackupUsersOnTargetHost) > 0 {
+		statement, err := dbagent.GenerateBackupForUsers(config.Config.MySQLBackupUsersOnTargetHost)
+		if err != nil {
+			return log.Errore(err)
+		}
+		f, err := os.OpenFile(path.Join(config.Config.MySQLBackupDir, mysqlUserBackupFileName), os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0666)
+		defer f.Close()
+		if err != nil {
+			return log.Errore(err)
+		}
+		_, err = f.WriteString(statement)
+		if err != nil {
+			return log.Errore(err)
+		}
+	} else {
+		cmd := fmt.Sprintf("mysqldump --master-data=2 --set-gtid-purged=OFF --user=%s --password=%s --port=%d --single-transaction mysql %s > %s",
+			config.Config.MySQLTopologyUser, config.Config.MySQLTopologyPassword, config.Config.MySQLPort, strings.Join(mysqlUsersTables[:], " "), path.Join(config.Config.MySQLBackupDir, mysqlUserBackupFileName))
```

Comment:
`mysqlUsersTables[:]` == `mysqlUsersTables`?

Reply:
fixed in 3ce673a